src/client/http/client.rs

```diff
@@ -76,21 +84,73 @@ impl HttpClient {
 		}
 	}
 
-	fn process_response(response: jsonrpc::Output, expected_id: u64) -> Result<JsonValue, Error> {
-		match response.id() {
-			jsonrpc::Id::Num(n) if n == &expected_id => {
-				let ret: Result<JsonValue, _> = response.into();
-				ret.map_err(Error::Request)
-			}
-			jsonrpc::Id::Num(n) => {
-				Err(Error::InvalidRequestId(Mismatch { expected: expected_id.into(), got: (*n).into() }))
+	/// Perform a batch request towards the server.
+	///
+	/// Returns `Ok` if all requests were answered successfully.
+	/// Returns `Error` if any of the requests fails.
+	///
+	/// If more than `u64::MAX` requests are performed in the `batch` then duplicate IDs are used
```

Comment:
Technically, I think iterators are capped at u32/u64 so this comment is probably faulty.